iothub/device/src/ClientFactory.cs

```diff
@@ -99,8 +118,7 @@ internal static InternalClient Create(string hostName, IAuthenticationMethod aut
         /// This initializer exists only for unit tests to override the pipeline creation
         /// </summary>
         /// <param name="pipelineBuilder">Should only be specified by SDK tests to override the operation pipeline.</param>
-        /// <param name="connectionString">The device connection string.</param>
-        /// <param name="authenticationMethod">The authentication method.</param>
```

Comment:
Changed this choke point factory method (last stop before calling `new` on `InternalClient` from taking a connection string. Now it takes the parsed `IotHubConnectionStringBuilder` class that Abhipsa will change to `DeviceIdentity`.

Consequently, isnce `IotHubConnectionStringBuilder` also has the `IAuthenticationMethod`, we don't need a separate param for it.